message-queue-lib/src/cmr/message_queue/topic/aws_topic.clj

```diff
@@ -47,26 +54,93 @@
             (recur (rest new-keys) (conj result {attr-key attr-value})))
           result)))))
 
+(defn subscribe-sqs-to-sns
+  "Subscribes an AWS SQS to an AWS Topic."
+  [sns-client topic-arn sqs-arn]
+  (let [sub-request (-> (SubscribeRequest/builder)
+                        (.protocol "sqs")
+                        (.endpoint sqs-arn)
+                        (.returnSubscriptionArn true)
+                        (.topicArn topic-arn)
+                        (.build))
+        response (.subscribe sns-client sub-request)]
+    (.subscriptionArn response)))
+
+(defn set-filter-policy
+  "For a given subscription set the filter policy so that the queue
+    only gets the notificiation messages that it wants. The passed in
+    filter policy is a hash map - for example:
+    {\"collection-concept-id\": \"C12345-PROV1\"
+     \"mode\": [\"New\", \"Update\"]}"
+  [sns-client subscription-arn subscription]
+  ;; Turn the clojure filter policy to json
+  (when (or (:CollectionConceptId subscription)
```

Comment:
If both :CollectionConceptId and :Mode are required, would it better to do an and?
Or does the presence of one guarantee the other?